core/src/main/java/cucumber/runtime/arquillian/client/CucumberArchiveProcessor.java

```diff
@@ -355,4 +329,38 @@ private static String featureName(final URL url) {
         // fallback
         return Math.abs(url.hashCode()) + Features.EXTENSION;
     }
+
+    private static boolean isCukeSpaceTest(final TestClass testClass, final ClassLoader loader) {
+        Class<? extends Annotation> runWithType = null;
+        try {
+            runWithType = (Class<? extends Annotation>) loader.loadClass("org.junit.runner.RunWith");
+
+        } catch (final ClassNotFoundException error) {
+            // no-op
+        } catch (final NoClassDefFoundError e) {
+            // no-op
+        }
+
+        if (runWithType == null) {
+            return false;
+        }
+        final Annotation runWith = testClass.getAnnotation(runWithType);
+        if (runWith == null) {
+            return false;
+        }
+
+        try {
+            final Class<?> runner = Class.class.cast(runWithType.getMethod("value").invoke(runWith));
+            if ((!ArquillianCucumber.class.equals(runner) && !CukeSpace.class.equals(runner))) {

```

Comment:
about the name: we don't use it as a processor but really an appender/enricher. We can't use the other arquillian SPI cause of a limitation but semantically it sounds more correct, wdyt?

about the helper: i'm not a fan of helper classes - actually I regret the ones we have - and would prefer composable services but also depends what it is, if very specific it can makes sense. Let's move this discussion to another PR maybe?